src/containers/classrooms/ClassroomsTableContainer.jsx

```diff
@@ -34,17 +39,40 @@ class ClassroomsTableContainer extends React.Component {
   }
 
   resetState() {
-    this.setState({ classroomToDelete: null, showConfirmationDialog: false });
+    this.setState({
+      assignmentToDelete: null,
+      classroomToDelete: null,
+      showConfirmationDialog: {
+        assignment: false,
+        classroom: false
+      }
+    });
+  }
+
+  maybeDeleteAssignment(id) {
+    this.setState({ assignmentToDelete: id, showConfirmationDialog: { assignment: true } });
```

Comment:
This `setState()` call appears to 'erase' the `state.showConfirmationDialog.classroom` key-value; i.e. replacing `showConfirmationDialog: { assignment: false, classroom: false }` with `showConfirmationDialog: { assignment: true }` .

That said, in practice I detect no issues with this, as `undefined` is functionally similar enough to `false` that there shouldn't be any hiccups.

Reply:
doh, gotta remember that setState doesn't deep merge.